src/qibocal/protocols/coherence/zeno_signal.py

```diff
@@ -55,6 +46,25 @@ class ZenoSignalResults(Results):
     """Approximate covariance of fitted parameters."""
 
 
+def zeno_sequence(
+    platform: Platform, targets: list[QubitId], readouts: int
+) -> tuple[PulseSequence, dict[QubitId, int]]:
+    """Generating sequence for Zeno experiment."""
+
+    sequence = PulseSequence()
+    readout_duration = {}
+    for q in targets:
+        natives = platform.natives.single_qubit[q]
+        _, ro_pulse = natives.MZ()[0]
+        readout_duration[q] = ro_pulse.duration
+        sequence |= natives.RX()
+
+        for _ in range(readouts):
+            sequence |= natives.MZ()
```

Comment:
I believe this is different than the corresponding 0.1. Before I see that `RX` had always `start=0`, so we were basically doing
```
q0: --- X --- M --- M --- M
q1: --- X --- M --- M --- M
```
while now it will do
```
q0: --- X --- M --- M --- M -----------------------
q1: --------------------------- X --- M --- M --- M
```
due to the way `PulseSequence` contatenation (`|`) works.

I think the easiest fix would something like
```suggestion
        qubit_sequence = natives.RX() | natives.MZ()
        for _ in range(readouts - 1):
            qubit_sequence += natives.MZ()
        sequence += qubit_sequence
```
but I have not tested.

Reply:
Yes, exactly.

At some point, I passed a confusing info (that I amended right after), regarding the measurements' identity, especially when they are added in a loop. Since measurements ID have to be unique, you should avoid adding the same variable multiple times in a sequence.
However, the key ingredient here was not the `|` (which is just a way of syncing, as the `<<` operator - one acting per-channel, the other per-qubit), but instead the call, i.e. the `MZ()` (aka `MZ.__call__()`) should be called within the loop, and not outside and stored in a variable.

I hope this information has been passed appropriately (btw @stavros11, we should definitely better document this part - maybe we could start opening an issue with items to be properly documented?)

Reply:
Thanks @stavros11, indeed before the tutorial I was aware of the difference between `|` and `+`